packages/core/src/format/codeframes.ts

```diff
@@ -183,8 +183,10 @@ export function getAstNodeByPointer(root: YAMLNode, pointer: string, reportOnKey
   for (const key of pointerSegments) {
     if (currentNode.kind === yamlAst.Kind.MAP) {
       const mapping = currentNode.mappings.find((m) => m.key.value === key);
-      if (!mapping?.value) break;
-      currentNode = mapping?.value as YAMLNode;
+      if (!mapping) break;
```

Comment:
Could we somehow avoid using the `break`s? Would extracting these checks into a separate function be an option?

Reply:
I think we will have the same picture but with `return`s instead of `break`s